firefox-ios/Client/Frontend/Browser/Toolbars/Models/NavigationToolbarContainerModel.swift

```diff
@@ -60,11 +60,20 @@ struct NavigationToolbarContainerModel: Equatable {
         } : nil
     }
 
+    /// Handles user action for tapping on one of the buttons on the navigation toolbar.
+    /// In the case of tapping on search, we want to handle it differently for when we show homepage search bar feature.
     private static func getOnSelected(action: ToolbarActionConfiguration, windowUUID: WindowUUID) -> ((UIButton) -> Void)? {
         return { button in
+            let shouldShowSearchBar = store.state.screenState(
```

Comment:
How about instead of adding logic here we dispatch a `zeroSearch` action which in the `ToolbarMiddleware` adds `isHomepageSearchBarEnabled` to the action dispatched. Would that work? Just would like to have the logic in one place if possible.

Reply:
When I think of our Redux architecture, when we dispatch an action, it's associated with a user action or API action and we don't care about the consequence. So dispatching a `zeroSearch` action seems that we're thinking about the consequence vs. the action that was taken. 

Unless you prefer that we handle this logic in `didTapButton` in `ToolbarMiddleware`, then would need to find a better name for the middleware action. No strong opinion on this but middlewares to me are for communicating to external dependencies that we inject and this is a simple logic check here with a global state.  

Happy to chat on coffee chat if I'm misunderstanding!